platform/o.n.swing.laf.flatlaf/src/org/netbeans/swing/laf/flatlaf/ui/FlatViewTabDisplayerUI.java

```diff
@@ -207,40 +220,51 @@ protected void paintTabBackground(Graphics g, int index, int x, int y, int width
     }
 
     private void paintTabBackgroundAtScale1x(Graphics2D g, int index, int width, int height, double scale) {
-        // do not round tab separator width to get nice small lines at 125%, 150% and 175%
-        int tabSeparatorWidth = (showTabSeparators && index >= 0) ? (int) (1 * scale) : 0;
+        boolean selected = isSelected(index);
 
-        // paint background
         Color bg = colorForState(index, background, activeBackground, selectedBackground,
-                hoverBackground, attentionBackground);
+                selectedHoverBackground, hoverBackground, attentionBackground);
+
+        boolean showSeparator = showTabSeparators &&
+                // Show separators _between_ tabs (not after the last one), like for editor tabs.
+                index >= 0 && index < getDataModel().size() - 1 &&
+                /* Don't show separators around the selected tab, if there's already a border or
+                color contrast at its sides. */
+                (!showSelectedTabBorder && selectedBackground.equals(activeBackground) ||
+                (!selected && !isSelected(index + 1)));
```

Comment:
thanks for documenting this. 
In case this would get longer in future this could make use of some self documenting utility methods:
```
boolean showSeparator = showTabSeparators && isNotLast(index) && hasLowContrastWithNeighbor(index);
```
But this is fine too IMO.

Reply:
I rewrote this into a function. The logic should be the same as before. (Rewriting (!a && !b) to !(a || b) a few times...)